### DISCUSSION TITLE
1.8.0

### DISCUSSION THREAD

**Discussion by ogoffart** [slint-ui/slint#6274] [Blog Comments]
[**📰 Release Blog**](https://slint.dev/blog/slint-1.8-released)

Slint 1.8.0 is a patch release that fixes several bugs and crashes. For a complete list of changes, check out the [ChangeLog](https://github.com/slint-ui/slint/blob/master/CHANGELOG.md#180---2024-09-23).

Upgrading to Slint 1.8.0:

   * Rust: Run `cargo update`.
   * C++: If you're using `FetchContent` with CMake, update the GIT_TAG entry in CMakeLists.txt to `v1.8.0` or `release/1.8` for automatic tracking of 1.8.x releases.
   * Node.js: Run `npm update slint-ui` to bring your `package.json` up-to-date.
   * ESP-IDF developers: Run `idf.py update-dependencies` to bring your idf_component.yml up-to-date.

# Assets

## C++ SDK

For C++ developers, this release contains the follwing pre-packed builds of Slint:

| Desktop Operating System | Compiler | Architecture | Download |
|--------------------------|----------|--------------|----------|
| Linux                    | gcc      | x86-64       | [Slint-cpp-1.8.0-Linux-x86_64.tar.gz](https://github.com/slint-ui/slint/releases/download/1.8.0/Slint-cpp-1.8.0-Linux-x86_64.tar.gz) |
| Windows                  | MSVC     | x86-64       | [Slint-cpp-1.8.0-win64-MSVC.exe](https://github.com/slint-ui/slint/releases/download/1.8.0/Slint-cpp-1.8.0-win64-MSVC.exe) |

| Templates for Microcontroller Evaluation Board | Download |
|----------------------------------|----------|
| [STM32H747I-DISCO](https://www.st.com/en/evaluation-tools/stm32h747i-disco.html): Dual-core Arm M7/M4 MCU with 4” touch LCD display module | [slint-cpp-template-stm32h747i-disco.zip](https://github.com/slint-ui/slint/releases/download/1.8.0/slint-cpp-template-stm32h747i-disco.zip) |
| [STM32H735G-DK](https://www.st.com/en/evaluation-tools/stm32h735g-dk.html): Arm M7 MCU with 4” touch LCD display module | [slint-cpp-template-stm32h735g-dk.zip](https://github.com/slint-ui/slint/releases/download/1.8.0/slint-cpp-template-stm32h735g-dk.zip) |

## Tool Binaries

| Add-on    | Linux x86-64 | Windows | macOS |
|-----------|--------------|---------|-------|
| **slint-lsp** | [slint-lsp-linux.tar.gz](https://github.com/slint-ui/slint/releases/download/1.8.0/slint-lsp-linux.tar.gz) | [slint-lsp-windows.zip](https://github.com/slint-ui/slint/releases/download/1.8.0/slint-lsp-windows.zip) | [slint-lsp-macos.tar.gz](https://github.com/slint-ui/slint/releases/download/1.8.0/slint-lsp-macos.tar.gz) |
| **slint-viewer** | [slint-viewer-linux.tar.gz](https://github.com/slint-ui/slint/releases/download/1.8.0/slint-viewer-linux.tar.gz) | [slint-viewer-windows.zip](https://github.com/slint-ui/slint/releases/download/1.8.0/slint-viewer-windows.zip) | [slint-viewer-macos.tar.gz](https://github.com/slint-ui/slint/releases/download/1.8.0/slint-viewer-macos.tar.gz) |

<hr /><em>This discussion was created from the release <a href='https://github.com/slint-ui/slint/releases/tag/v1.8.0'>1.8.0</a>.</em>

**Comment by Enyium** [comment]
Congratulations! `Timer` will be very useful.

> Math Gains Postfix Support
> [...] Less intuitive for operations like clamp

Why should it be less intuitive? When I first read the article, I thought: "Oh, `clamp()` must've been defined with an argument order not suitable for the postfix afterthought." But now, I checked in the docs and via testing, and am wondering why you would say that. `percentage.clamp(0, 100)` looks perfectly fine to me. Rust [also](https://doc.rust-lang.org/std/primitive.f64.html#method.clamp) uses this approach.

> So for now you cannot use postfix for all functions in the Math namespace.

This made me think you made it a hard error to use `clamp()` that way. But this isn't the case.
